attachment_synchronize_pattimpex/models/attachment_queue.py

```diff
@@ -0,0 +1,51 @@
+# Copyright 2020 Akretion (http://www.akretion.com).
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import _, fields, models
+
+
+class AttachmentQueue(models.Model):
+    _inherit = "attachment.queue"
+
+    export_id = fields.Many2one("ir.exports")
+    file_type = fields.Selection(
+        selection_add=[("import_pattern", "Import using Patterns")]
+    )
+    pattimpex_id = fields.Many2one(
+        "patterned.import.export",
+        readonly=True,
+        string="Patterned Import/Export",
+        help="The Patterned Import/Export (and its import job) that was created",
+    )
+
+    # Import part
+
+    def _run_import_pattern(self):
+        patterned_import = self.env["patterned.import.export"].create(
+            {
+                "name": self.datas_fname,
+                "attachment_id": self.attachment_id.id,
+                "kind": "import",
+            }
+        )
+        self.pattimpex_id = patterned_import
+        description = _(
+            "Generate import '{model}' with pattern '{export_name}' using "
+            "format {format}"
+        ).format(
+            model=self.export_id.model_id.model,
+            export_name=self.export_id.name,
+            format=self.export_id.export_format,
+        )
+        self.export_id.with_delay(
```

Comment:
I would not delay it, since attachment.queue is already a queue system.
Less good thant queue.job indeed but still, I don't think it is necessary. 
If one attachment.queue fails, it does not block the others, a commit is done after each process of a file, etc...

I am not sure, but for the end user, it may become kind of complicated no?